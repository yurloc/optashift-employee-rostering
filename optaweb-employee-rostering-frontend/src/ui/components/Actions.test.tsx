/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { SizeMeProps } from 'react-sizeme';
import { Props, Actions } from './Actions';

describe('Actions component', () => {
  it('should render as all buttons when given enough space', () => {
    const actionsComponent = shallow(<Actions {...desktopProps} />);
    expect(toJson(actionsComponent)).toMatchSnapshot();
  });

  it('should render as both buttons and dropdown when given some space', () => {
    const actionsComponent = shallow(<Actions {...tabletProps} />);
    expect(toJson(actionsComponent)).toMatchSnapshot();
  });

  it('should render as all dropdown when given no space', () => {
    const actionsComponent = shallow(<Actions {...mobileProps} />);
    expect(toJson(actionsComponent)).toMatchSnapshot();
  });

  it('clicking on a button should call the action', () => {
    const actionsComponent = shallow(<Actions {...desktopProps} />);
    actionsComponent.find('Button[aria-label="Action 1"]').simulate('click');
    expect(actions[0].action).toBeCalled();
  });

  it('clicking on a dropdown should call the action', () => {
    const actionsComponent = shallow(<Actions {...mobileProps} />);
    actionsComponent.find('Dropdown').simulate('select', { currentTarget: { innerText: 'Action 3' } });
    expect(actions[2].action).toBeCalled();
  });
});

const actions = [
  { name: 'Action 1', action: jest.fn() },
  { name: 'Action 2', action: jest.fn() },
  { name: 'Action 3', action: jest.fn() },
];

const desktopProps: Props & SizeMeProps = {
  actions,
  size: {
    width: 2000,
    height: 1500,
  },
};

const tabletProps: Props & SizeMeProps = {
  actions,
  size: {
    width: 400,
    height: 600,
  },
};

const mobileProps: Props & SizeMeProps = {
  actions,
  size: {
    width: 100,
    height: 150,
  },
};
