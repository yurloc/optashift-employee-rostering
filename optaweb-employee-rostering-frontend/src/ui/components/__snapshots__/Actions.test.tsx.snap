// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions component should render as all buttons when given enough space 1`] = `
<span
  style={
    Object {
      "display": "grid",
      "gridTemplateColumns": "1fr auto",
    }
  }
>
  <span
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <span>
    <Button
      aria-label="Action 1"
      key="Action 1"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "5px",
        }
      }
    >
      Action 1
    </Button>
    <Button
      aria-label="Action 2"
      key="Action 2"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "5px",
        }
      }
    >
      Action 2
    </Button>
    <Button
      aria-label="Action 3"
      key="Action 3"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "5px",
        }
      }
    >
      Action 3
    </Button>
  </span>
</span>
`;

exports[`Actions component should render as all dropdown when given no space 1`] = `
<span
  style={
    Object {
      "display": "grid",
      "gridTemplateColumns": "1fr auto",
    }
  }
>
  <span
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <span>
    <Dropdown
      dropdownItems={
        Array [
          <DropdownItem>
            Action 1
          </DropdownItem>,
          <DropdownItem>
            Action 2
          </DropdownItem>,
          <DropdownItem>
            Action 3
          </DropdownItem>,
        ]
      }
      isOpen={false}
      isPlain={true}
      onSelect={[Function]}
      position="right"
      toggle={
        <KebabToggle
          onToggle={[Function]}
        />
      }
    />
  </span>
</span>
`;

exports[`Actions component should render as both buttons and dropdown when given some space 1`] = `
<span
  style={
    Object {
      "display": "grid",
      "gridTemplateColumns": "1fr auto",
    }
  }
>
  <span
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <span>
    <Button
      aria-label="Action 1"
      key="Action 1"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "5px",
        }
      }
    >
      Action 1
    </Button>
    <Button
      aria-label="Action 2"
      key="Action 2"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "5px",
        }
      }
    >
      Action 2
    </Button>
    <Dropdown
      dropdownItems={
        Array [
          <DropdownItem>
            Action 3
          </DropdownItem>,
        ]
      }
      isOpen={false}
      isPlain={true}
      onSelect={[Function]}
      position="right"
      toggle={
        <KebabToggle
          onToggle={[Function]}
        />
      }
    />
  </span>
</span>
`;
